### PR TITLE
DHFPROD-7356:Skip RTL and Cypress tests on BE PR's

### PR DESCRIPTION
**Pull request by yafeldman**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by Sanjeevani19 — COMMENTED** [review]
@yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?

File: Jenkinsfile [hunk 1]
```diff
@@ -596,6 +598,19 @@ void mergePR(){
     }
 }
 
+def isChangeInUI(){
```
Comment (SameeraPriyathamTadikonda) [inline Jenkinsfile:601]: We don't need a new method to check the diffs.
We can use Jenkins OOTB functions something similar to check if the changes are in UI folder
`when {
                            changeset '**/UI/**'
                   }`

**Review by SameeraPriyathamTadikonda — CHANGES_REQUESTED** [review]

**Comment by yafeldman** [comment]
> @yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?

Why do you think 

> @yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?

@Sanjeevani19 I am not sure if we want to get rid of RTL tests because:
  1) only cypress test causing the most issues and backeend team originally was asking only get rid of cypress tests
  2) we might want to keep them as another testing backend changes 
  3) it requires to change current gradle tasks configuration

**Comment by Sanjeevani19** [comment]
> > @yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?
> 
> Why do you think
> 
> > @yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?
> 
> @Sanjeevani19 I am not sure if we want to get rid of RTL tests because:
> 
> 1. only cypress test causing the most issues and backeend team originally was asking only get rid of cypress tests
> 2. we might want to keep them as another testing backend changes
> 3. it requires to change current gradle tasks configuration

@yafeldman during sprint review meeting the team came to an agreement on skipping RTL unit tests too along with cypress since we are running them with mock data and BE changes wouldn't impact RTL tests, I am fine with the current approach of skipping only cypress tests for now if it involves too many changes.

**Comment by ngodugu-marklogic** [comment]
> > > @yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?
> > 
> > 
> > Why do you think
> > > @yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?
> > 
> > 
> > @Sanjeevani19 I am not sure if we want to get rid of RTL tests because:
> > 
> > 1. only cypress test causing the most issues and backeend team originally was asking only get rid of cypress tests
> > 2. we might want to keep them as another testing backend changes
> > 3. it requires to change current gradle tasks configuration
> 
> @yafeldman during sprint review meeting the team came to an agreement on skipping RTL unit tests too along with cypress since we are running them with mock data and BE changes wouldn't impact RTL tests, I am fine with the current approach of skipping only cypress tests for now if it involves too many changes.

@yafeldman As @Sanjeevani19 mentioned the agreement was to skip both RTL and Cypress on BE PRs

File: Jenkinsfile (hunk 1 above)
Comment (yafeldman) [reply Jenkinsfile:601]: I tested using `when { changeset '**/UI/**' }` and it doesn't seem like is doing what we need. 

And from Jenkins doc the  changeset parameter: `Execute the stage if the build’s SCM changeset contains one or more files matching the given pattern ` It is not clear to me what SCM changeset, but seem like is not changes of underline PR.

**Comment by yafeldman** [comment]
> > > > @yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?
> > > 
> > > 
> > > Why do you think
> > > > @yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?
> > > 
> > > 
> > > @Sanjeevani19 I am not sure if we want to get rid of RTL tests because:
> > > 
> > > 1. only cypress test causing the most issues and backeend team originally was asking only get rid of cypress tests
> > > 2. we might want to keep them as another testing backend changes
> > > 3. it requires to change current gradle tasks configuration
> > 
> > 
> > @yafeldman during sprint review meeting the team came to an agreement on skipping RTL unit tests too along with cypress since we are running them with mock data and BE changes wouldn't impact RTL tests, I am fine with the current approach of skipping only cypress tests for now if it involves too many changes.
> 
> @yafeldman As @Sanjeevani19 mentioned the agreement was to skip both RTL and Cypress on BE PRs

@ngodugu-marklogic I see @Sanjeevani19 is OK to skip only Cypress tests. But if we are still confident in skipping RTL then I'd need confirmation and ,probably implementation,  from the team for changes in current behavior in gradle tasks. Because, unless I am missing something, the entire HC test gradle task is finalizedBy RTL test , which means currently HC test always run RTL tests at the end.

File: Jenkinsfile (hunk 1 above)
Comment (SameeraPriyathamTadikonda) [reply Jenkinsfile:601]: @yafeldman It is working fine in other pipelines. Please let me know the scenario in which it did not work so I can test it on other pipelines.

SCM changeset is the set of files that changed in a commit (in our case its a PR).

**Comment by SameeraPriyathamTadikonda** [comment]
> > > > > @yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?
> > > > 
> > > > 
> > > > Why do you think
> > > > > @yafeldman Looks like pipeline is still running RTL unit tests, can you check once again ?
> > > > 
> > > > 
> > > > @Sanjeevani19 I am not sure if we want to get rid of RTL tests because:
> > > > 
> > > > 1. only cypress test causing the most issues and backeend team originally was asking only get rid of cypress tests
> > > > 2. we might want to keep them as another testing backend changes
> > > > 3. it requires to change current gradle tasks configuration
> > > 
> > > 
> > > @yafeldman during sprint review meeting the team came to an agreement on skipping RTL unit tests too along with cypress since we are running them with mock data and BE changes wouldn't impact RTL tests, I am fine with the current approach of skipping only cypress tests for now if it involves too many changes.
> > 
> > 
> > @yafeldman As @Sanjeevani19 mentioned the agreement was to skip both RTL and Cypress on BE PRs
> 
> @ngodugu-marklogic I see @Sanjeevani19 is OK to skip only Cypress tests. But if we are still confident in skipping RTL then I'd need confirmation and ,probably implementation, from the team for changes in current behavior in gradle tasks. Because, unless I am missing something, the entire HC test gradle task is finalizedBy RTL test , which means currently HC test always run RTL tests at the end.

@yafeldman I implemented the gradle task and you can remove the condition if team decided to skip running rtl tests  on a PR.

**Comment by yafeldman** [comment]
> > if team decided to skip running rtl tests on a PR.
> 
> @yafeldman Do you have all the info needed to skip RTL tests ? let's skip both RTL and cypress tests as decided before

Yes pretty much. Thank you for clarifications. Still 2 last questions:
 1)  @ngodugu-marklogic Are RTL tests are independent from the rest of HC tests? and can be run as separate suit?
  2) @Sanjeevani19 just to confirm  if 1) above is correct can I run these tests as independent stage in DHF pipeline if PR is  UI change?

**Comment by Sanjeevani19** [comment]
> > > if team decided to skip running rtl tests on a PR.
> > 
> > 
> > @yafeldman Do you have all the info needed to skip RTL tests ? let's skip both RTL and cypress tests as decided before
> 
> Yes pretty much. Thank you for clarifications. Still 2 last questions:
> 
> 1. @ngodugu-marklogic Are RTL tests are independent from the rest of HC tests? and can be run as separate suit?
> 2. @Sanjeevani19 just to confirm  if 1) above is correct can I run these tests as independent stage in DHF pipeline if PR is  UI change?

@yafeldman yes, RTL tests are independent from the rest of HC tests and can be ran independently on the pipeline

File: Jenkinsfile (hunk 1 above)
Comment (yafeldman) [reply Jenkinsfile:601]: I just added fake ui change and added `changeset "**/ui/**"` and run build: https://jenkins.marklogic.com/blue/organizations/jenkins/Datahub_CI/detail/PR-5709/16/pipeline/87

and it still skips cypress stage

**Review by SameeraPriyathamTadikonda — APPROVED** [review]

**Review by akshaysonvane — DISMISSED** [review]

**Review by ngodugu-marklogic — APPROVED** [review]

**Review by rahulvudutala — APPROVED** [review]

**Pull request merged**